core/src/subgraph/context/instance.rs

```diff
@@ -111,6 +116,11 @@ where
                 sender
             }
         };
+
+        if let Some(address) = data_source.address() {
+            self.host_addresses.insert(address.into());
```

Comment:
I am not sure if I am missing something, but I don't see `self.host_addresses` used anywhere to check if an address is there or not.

Reply:
Oh wow seems I completely lost track of where I was going with this 😅. It's not only the AIs that can hallucinate. I'll rework this to actually use the hashset.